tests/scripts/all.sh

```diff
@@ -1579,7 +1579,7 @@ component_test_no_64bit_multiplication () {
 component_build_arm_none_eabi_gcc () {
     msg "build: ${ARM_NONE_EABI_GCC_PREFIX}gcc, make" # ~ 10s
     scripts/config.py baremetal
-    make CC="${ARM_NONE_EABI_GCC_PREFIX}gcc" AR="${ARM_NONE_EABI_GCC_PREFIX}ar" LD="${ARM_NONE_EABI_GCC_PREFIX}ld" CFLAGS='-Werror -Wall -Wextra' lib
+    make CC="${ARM_NONE_EABI_GCC_PREFIX}gcc" AR="${ARM_NONE_EABI_GCC_PREFIX}ar" LD="${ARM_NONE_EABI_GCC_PREFIX}ld" CFLAGS='-Werror -Wall -Wextra -O1' lib
```

Comment:
Just wondering, why not -O2? That's the level of optimization I am used to for release builds.

Reply:
This is only a build, it won't run, so `-O2` has no benefit over `-O1`, and it's slower. `-Oanything_non_zero` has the benefit that it activates some compile-time analyses that gcc doesn't try at `-O0`.

Reply:
I understand it doesn't run. In my humble Cortex-M experience, -O1 is not really used. -O0 is used for debug builds and then -O2 or -Os are used for "release" builds. Thus if we are at building with some optimization it is maybe a bit better to choose an optimization that is more likely to be used. There are also probably more compile-time analyses when using -O2 compared to -O1. The additional compilation time should be negligible. Thus I see some benefits of choosing -O2 over -O1 here. But if you are not convinced by the above, fine by me, just let me know and I will approve the PR.

Reply:
The main purpose of this change was actually to build the assembly code that we gate with `defined(__OPTIMIZE__)`. I don't see a need to make the build slower here. That being said, as Manuel said, the arm builds need further rationalizing, but that's beyond the scope of this PR. I've filed https://github.com/ARMmbed/mbedtls/issues/3299. I don't think any further change is warranted in this PR.

Reply:
ok I wasn't aware of "defined(__OPTIMIZE__)". Thanks for #3299, this looks very good to me.